src/components/StaffList/index.jsx

```diff
@@ -10,10 +12,56 @@ class StaffList extends React.Component {
     super(props);
     this.state = {
       staffList: props.input.value || [], // requires staffList to be an array
+      suggestions: [], // the suggestions returned by the discovery service for autocomplete
+      searchString: '', // search input provided by the user
     };
 
     this.handleRemove = this.handleRemove.bind(this);
     this.onDragEnd = this.onDragEnd.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this);
+    this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(this);
+    this.getSuggestionValue = this.getSuggestionValue.bind(this);
+    this.renderSuggestion = this.renderSuggestion.bind(this);
+  }
+
+  onChange(event, { newValue }) {
+    this.setState({ searchString: newValue });
+  }
+
+  /**
+   * The callback method for when a user has provided input in the search input field for
+   * autocomplete.  Responsible for deciding if we need to fetch new results (if input is long
+   * enough).
+   *
+   * @param value - the text the user has input in the search input element.
+   * @return Promise - continue async execution
+   */
+  onSuggestionsFetchRequested({ value }) {
+    if (value.length >= 2) { // We don't hit the discovery service until we have enough characters
+      return DiscoveryDataApiService.autocompletePerson(value, null)
```

Comment:
super nit: I like the below format more when chaining... but it's mostly personal preference
```javascript
return stuff
  .function()
  .anotherFunction()
```

Reply:
👍